satpy/readers/amsr2_l2_gaasp.py

```diff
@@ -0,0 +1,286 @@
+        }
+        x_dim_name = data_arr.dims[1]
+        if x_dim_name in self.dim_resolutions:
+            ds_info['resolution'] = self.dim_resolutions[x_dim_name]
+        if not self.is_gridded and data_arr.coords:
+            self._add_lonlat_coords(data_arr, ds_info)
+        return ds_info
+
+    def _available_new_datasets(self):
+        handled_variables = set()
+        possible_vars = list(self.nc.data_vars.items()) + list(self.nc.coords.items())
+        for var_name, data_arr in possible_vars:
+            if var_name in handled_variables:
+                continue
+            if data_arr.ndim != 2:
+                # we don't currently handle non-2D variables
+                continue
+            has_y_dim = data_arr.dims[0] in self.y_dims
+            has_x_dim = data_arr.dims[1] in self.x_dims
+            if not has_y_dim or not has_x_dim:
+                # we need 'traditional' y/x dimensions currently
+                continue
+
+            ds_info = self._get_ds_info_for_data_arr(var_name, data_arr)
+            handled_variables.add(var_name)
+            yield True, ds_info
+
+    def available_datasets(self, configured_datasets=None):
+        """Dynamically discover what variables can be loaded from this file.
+
+        See :meth:`satpy.readers.file_handlers.BaseHandler.available_datasets`
+        for more information.
+
+        """
+        yield from self._available_if_this_file_type(configured_datasets)
+        yield from self._available_new_datasets()
+
+
+class GAASPGriddedFileHandler(GAASPFileHandler):
+    """GAASP file handler for gridded products like SEAICE."""
+
+    y_dims = (
+        'Number_of_Y_Dimension',
+    )
+    x_dims = (
+        'Number_of_X_Dimension',
+    )
+    dim_resolutions = {
+        'Number_of_X_Dimension': 10000,
+    }
+    is_gridded = True
+
+    @staticmethod
+    def _get_extents(data_shape, res):
+        # assume data is centered at projection center
+        x_min = -(data_shape[1] / 2.0) * res
+        x_max = (data_shape[1] / 2.0) * res
+        y_min = -(data_shape[0] / 2.0) * res
+        y_max = (data_shape[0] / 2.0) * res
+        return x_min, y_min, x_max, y_max
+
+    def get_area_def(self, dataid):
+        """Create area definition for equirectangular projected data."""
+        var_suffix = self.filetype_info.get('var_suffix', '')
+        area_name = 'gaasp{}'.format(var_suffix)
+        orig_var_name = self._get_var_name_without_suffix(dataid['name'])
+        data_shape = self.nc[orig_var_name].shape
+        crs = CRS(self.filetype_info['grid_epsg'])
+        res = dataid['resolution']
+        extent = self._get_extents(data_shape, res)
+        area_def = AreaDefinition(
+            area_name,
+            area_name,
+            area_name,
+            crs,
+            data_shape[1],
+            data_shape[0],
+            extent
+        )
+        return area_def
+
+
+class GAASPLowRezFileHandler(GAASPFileHandler):
```

Comment:
LowRes vs LowRez? I see Res used more often, could we stick to it?

Reply:
Yes, good point, I will fix this.